homework-submission/React_HW2/todoApp2.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+// ContentDynamicTodo component is for map function to use
+const ContentDynamicTodo = ({ description, deadline, className, deleteItem }) => {
+  return (
+    <tbody>
+      <tr className={className} onClick={deleteItem}>
+        <td class="evenBackgroundColor">
+          <td class="content">{description}</td>
+          <td class="content">{deadline}</td>
+          <td class="content">{className}</td>
+        </td>
+      </tr>
+    </tbody>
+  );
+};
+
+// DynamicList where map function is
+const DynamicList = ({ list, deleteItem }) => {
+  const runTodoList = list.map(todo => (
+    <ContentDynamicTodo
+      key={todo.id}
+      description={todo.description}
+      deadline={todo.deadline}
+      className={todo.done ? '√' : 'x'}
+      deleteItem={() => {
+        deleteItem(todo.id);
+      }}
+    />
+  ));
+  return <tbody>{runTodoList}</tbody>;
+};
+
+// Here is our main Application component as a class
+class AppTodo2 extends React.Component {
+  //TodoList which contains new item as state
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+      newTodo: {
+        id: '',
+        description: 'Description',
+        deadline: 'No Deadline Entry',
+        done: false,
+      },
+    };
+  }
+
+  // Adding a description for new todo.
+  NewTodoDescription = () => {
```

Comment:
I see double,
starting from line 70 ending with line 87, is the exact same as starting from line 90 ending with line 109.

The only difference in these two components is one word in line 68 vs line 86 _(description)_ changed to _(deadline)_.

In any kind like of this situation, remember to dry **(Don't repeat yourself)**
to solve this you can do the following:
```javascript
  NewListItem = (props) => { // since we are using props here too!
    return (
      <p>
        <label htmlFor="">Deadline </label>
        <div id="containerDateInput">
          {' '}
          <input
            id="inputDate"
            type="date"
            name={props.name} // since this is a component we can specify the name via props
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                 // using the name in this kind of use allowing use to modify the key name to be as we specify the value of it
                newTodo: { ...this.state.newTodo, deadline: inputValue, id: Math.random() },
              });
            }}
          />
        </div>
      </p>
    );
  };
```
Example use of changed key name:
 name is 'description' means our object will look like: {..., description: 'somevalue'}
 same if 'deadline' means our object will look like: {..., deadline: 'some deadline'}

**Don't forgot to pass the name through props**

Reply:
that is what you mean
 NewTodoDescriptionAndDeadline = () => {
    return (
      <div>
      <div>
        <label htmlFor="">Description </label>
        <div id="containerDescriptionInput">
          <input
            type="text"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, description: inputValue, id: Math.random() },
              });
            }}
          />
        </div>
        </div>
        <div>
        <label htmlFor="">Deadline </label>
        <div id="containerDateInput">
          {' '}
          <input
            id="inputDate"
            type="date"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, deadline: inputValue },
              });
            }}
          />
        </div>
        </div>
        </div>
    );
  };

Reply:
that is what you mean.

Reply:
` NewTodoDescriptionAndDeadline = () => {
    return (
      <div>
      <div>
        <label htmlFor="">Description </label>
        <div id="containerDescriptionInput">
          <input
            type="text"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, description: inputValue, id: Math.random() },
              });
            }}
          />
        </div>
        </div>
        <div>
        <label htmlFor="">Deadline </label>
        <div id="containerDateInput">
          {' '}
          <input
            id="inputDate"
            type="date"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, deadline: inputValue },
              });
            }}
          />
        </div>
        </div>
        </div>
    );
  };`

Reply:
`
 NewTodoDescriptionAndDeadline = () => {
    return (
      <div>
      <div>
        <label htmlFor="">Description </label>
        <div id="containerDescriptionInput">
          <input
            type="text"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, description: inputValue, id: Math.random() },
              });
            }}
          />
        </div>
        </div>
        <div>
        <label htmlFor="">Deadline </label>
        <div id="containerDateInput">
          {' '}
          <input
            id="inputDate"
            type="date"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, deadline: inputValue },
              });
            }}
          />
        </div>
        </div>
        </div>
    );
  };
`

Reply:
NewTodoDescriptionAndDeadline = () => {
    return (
      <div>
      <div>
        <label htmlFor="">Description </label>
        <div id="containerDescriptionInput">
          <input
            type="text"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, description: inputValue, id: Math.random() },
              });
            }}
          />
        </div>
        </div>
        <div>
        <label htmlFor="">Deadline </label>
        <div id="containerDateInput">
          {' '}
          <input
            id="inputDate"
            type="date"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, deadline: inputValue },
              });
            }}
          />
        </div>
        </div>
        </div>
    );
  };

Reply:
```suggestion
  NewTodoDescription = () => {
```

Reply:
```suggestion
   NewTodoDescriptionAndDeadline = () => {
    return (
      <div>
      <div>
        <label htmlFor="">Description </label>
        <div id="containerDescriptionInput">
          <input
            type="text"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, description: inputValue, id: Math.random() },
              });
            }}
          />
        </div>
        </div>
        <div>
        <label htmlFor="">Deadline </label>
        <div id="containerDateInput">
          {' '}
          <input
            id="inputDate"
            type="date"
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                newTodo: { ...this.state.newTodo, deadline: inputValue },
              });
            }}
          />
        </div>
        </div>
        </div>
    );
  };NewTodoDescription = () => {
```

Reply:
Oh yeah, my bad, How it ended ip like that, it should be as:
  NewListItem = (props) => { // since we are using props here too!
    return (
      <p>
        <label htmlFor="">Deadline </label>
        <div id="containerDateInput">
          {' '}
          <input
            id="inputDate"
            type="date"
            name={props.name} // since this is a component we can specify the name via props
            onChange={event => {
              const inputValue = event.target.value;
              this.setState({
                 // using the name in this kind of use allowing use to modify the key name to be as we specify the value of it
                newTodo: { ...this.state.newTodo, [props.name]: inputValue, id: Math.random() },
              });
            }}
          />
        </div>
      </p>
    );
  };
```

your latest suggested changes are just making all the cattle in the same place, making it very illegible.